tests/unit/test_client.py

```diff
@@ -467,6 +467,14 @@ def test_client_validates_params(self):
             # Missing required 'Foo' param.
             service_client.test_operation(Bar='two')
 
+    def test_client_with_custom_timeout(self):
+        self.create_client_creator().create_client('myservice', 'us-west-2',
+            client_config=client.Config(connect_timeout=123, read_timeout=234))
```

Comment:
This is testing when different `connect/read` timeouts are provided.  Can we add a test for when only a read_timeout (or connect_timeout) is specified?